phylogenetic/rules/subsample.smk

```diff
@@ -22,6 +22,7 @@ rule subsample:
     input:
         sequences = input_sequences,
         metadata = input_metadata,
+        config = RUN_CONFIG,
```

Comment:
Huh, I was under this impression this would not work either because the file modification date will always be updated and be newer than the timestamp of the output (based on [Ignoring timestamps docs](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#ignoring-timestamps)). However, local testing shows this does work as expected.

Reply:
Ah, thanks for the pointer. This bit is important

> Hash input files that are not generated by other jobs (sorted by hash value).

So this works as expected because `RUN_CONFIG` is not an output of another rule, further supporting the decision to _not_ write the config as part of a Snakemake rule 😄